src/libs/Network/SequentialQueue.ts

```diff
@@ -360,25 +423,35 @@ function waitForIdle(): Promise<unknown> {
  * This is to prevent previous requests interfering with other tests
  */
 function resetQueue(): void {
+    const wasProcessingRelevantRequest = currentRequestCommand !== null;
     isSequentialQueueRunning = false;
     currentRequestPromise = null;
+    currentRequestCommand = null;
     isQueuePaused = false;
     isReadyPromise = new Promise((resolve) => {
         resolveIsReadyPromise = resolve;
     });
     resolveIsReadyPromise?.();
+
+    // Clear all listeners and notify of state change if needed
+    queueStateListeners.clear();
+    if (wasProcessingRelevantRequest) {
+        notifyQueueStateChange();
+    }
```

Comment:
I think the `notifyQueueStateChange` after the clearing doesn't make much sense, because the `queueStateListeners` will be always empty